internal/clients/team4/client.go

```diff
@@ -243,3 +245,64 @@ func (c *client) VoteForElection(roleToElect shared.Role, candidateList []shared
 
 	return returnList
 }
+
+
+//MonitorIIGORole decides whether to perform monitoring on a role
+//COMPULOSRY: must be implemented
+func (c *client) MonitorIIGORole(roleName shared.Role) bool {
+	
+	presidentID := c.getPresident()
+	speakerID := c.getSpeaker()
+	judgeID := c.getJudge()
+	clientID := shared.ClientID(4)
+	// TODO: Choose sensible thresholds!
+	trustThreshold := 0.5
+	resourcesThreshold := shared.Resources(100)
+	monitoring := false
+	if presidentID == clientID {
+		// If we are the president.
+		monitoring = (c.internalParam.agentsTrust[speakerID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)		   
+	} else if speakerID == clientID {
+		// If we are the Speaker.
+		monitoring = (c.internalParam.agentsTrust[presidentID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)
+	} else if judgeID == clientID {
+		// If we are the Judge.
+		monitoring = (c.internalParam.agentsTrust[speakerID] < trustThreshold ||
+					  c.internalParam.agentsTrust[judgeID]   < trustThreshold ) && 
+					 (c.ServerReadHandle.GetGameState().ClientInfo.Resources > resourcesThreshold)
+	}
+	return monitoring
+}
+
+//DecideIIGOMonitoringAnnouncement decides whether to share the result of monitoring a role and what result to share
+//COMPULSORY: must be implemented
+func (c *client) DecideIIGOMonitoringAnnouncement(monitoringResult bool) (resultToShare bool, announce bool) {
+	collaborationThreshold := 0.5
+	importance := mat.NewVecDense(5, []float64{
+		0.0, 1.0, -1.0, 1.0, 0.0,
+	})
```

Comment:
same as below about golden numbers. 

if we have some parameters, which we don't use, instead of adding 0 weight to them, we can just skip them altogether.

```
importance := mat.NewVecDense(3, []float64{
		1.0, -1.0, 1.0,
	})
```
and 
```
parameters := mat.NewVecDense(3, []float64{
		c.internalParam.selfishness,
		c.internalParam.fairness,
		c.internalParam.collaboration,
	})
```